cpp/src/arrow/compute/function.h

```diff
@@ -297,10 +299,10 @@ class ARROW_EXPORT ScalarFunction : public detail::FunctionImpl<ScalarKernel> {
   using KernelType = ScalarKernel;
 
   ScalarFunction(std::string name, const Arity& arity, FunctionDoc doc,
-                 const FunctionOptions* default_options = NULLPTR, bool is_impure = false)
+                 const FunctionOptions* default_options = NULLPTR, bool is_pure = true)
```

Comment:
I like this modification, but I also wonded that, would this affect legacy user?

Reply:
Oh, lucky, we don't have legacy user

Reply:
Yep. This was a very recent addition.